include/bm_baselineGB.hpp

```diff
@@ -0,0 +1,591 @@
+                destroy(E);
+            }
+            else
+            {
+                if (pos)
+                {
+                    // printf("DEBUG: clow_row !ID pos\n");
+                    S = mult1(row, ID, A, full_side);
+                    // GrB_Matrix_nvals(&elems, S);
+                }
+
+                else
+                {
+                    // printf("DEBUG: clow_row !ID !pos\n");
+                    // GrB_Index nvalsID;
+                    // GrB_Matrix_nvals(&nvalsID,ID);
+                    // printf("DEBUG:  ID elems in closrow: %d\n", nvalsID);
+                    E = empty(dim, dim);
+                    S = sum1(row, ID, E, full_side);
+                    destroy(E);
+                }
+            }
+            // GrB_Matrix_nvals(&elems, S);
+            bool accessS;
+            if (coltest)
+                GrB_Matrix_extractElement_BOOL(&accessS, S, row, *coltest);
+            if (coltest && accessS)
+            {
+                destroy(S);
+                *coltest = 1;
+                return NULL;
+            }
+            P = mult(S, A);
+            M = S;
+            S = sum(P, S);
+
+            GrB_Index elems_new;
+            GrB_Matrix_nvals(&elems_new, S);
+
+            while (elems_new != elems)
+            {
+                if (coltest)
+                    GrB_Matrix_extractElement_BOOL(&accessS, S, row, *coltest);
+                if (coltest && accessS)
+                {
+                    destroy(S);
+                    destroy(P);
+                    *coltest = 1;
+                    return NULL;
+                }
+                elems = elems_new;
+                M = P;
+                P = mult(P, A);
+                destroy(M);
+                M = S;
+                S = sum(S, P);
+                destroy(M);
+                GrB_Matrix_nvals(&elems_new, S);
+            }
+            destroy(P);
+            if (coltest)
+            {
+                destroy(S);
+                *coltest = 0;
+                return NULL;
+            }
+            // printf("DEBUG: clos_row before return\n");
+            return S;
+        }
+
+        static inline GrB_Matrix clos1(uint64_t row, GrB_Matrix A, uint pos, uint64_t col)
+        {
+            GrB_Index nrow, ncol;
+            // printf("DEBUG: clos1\n");
+            GrB_Index nrowsA, ncolsA;
+            GrB_Matrix_nrows(&nrowsA, A);
+            GrB_Matrix_ncols(&ncolsA, A);
+            uint side = mmax(nrowsA, ncolsA);
+            uint test;
+            GrB_Matrix M;
+            GrB_Matrix At;
+
+            if (row == full_side)
+            {
+                if (col == full_side)
+                    return clos(A, pos);
+                else
+                {
+                    GrB_Index n;
+                    At = transpose(A);
+                    M = clos_row(col, NULL, At, pos, NULL);
+                    M = transpose(M);
+                    return M;
```

Comment:
memory leak, i think

```suggestion
                    MT = transpose(M);
                    destroy(M);
                    return MT;
```